### DISCUSSION TITLE
텔레포트 기능이 작동하지 않아요.

### DISCUSSION THREAD

**Discussion by choramha** [naverz/zepeto-studio-kor#884] [5. Q&A]
월드 1.1.5 이용중입니다.
버튼을 누르면 원하는 위치에서 캐릭터가 모션을 시작하도록 만들고 싶습니다. 하지만 텔레포트 기능을 넣으면 이런 오류가 뜹니다. 어떻게 해결해야하나요? 

![image](https://user-images.githubusercontent.com/106102485/173523162-db06977d-668e-4c1e-a63e-263a0c86fb22.png)

![image](https://user-images.githubusercontent.com/106102485/173523022-64cae23a-0884-431b-bc63-224c2a5d9c65.png)

**Comment by YSHFM** [comment]
this.zepetoCharacter가 private이면 따로 Setter가 없는 것으로보입니다만 null이 아니신지...?

**Reply by Zepeto-TechEvangelist** [reply]
안녕하세요. 윗 분 말씀처럼 제페토 캐릭터가 null이어서 텔레포트 함수를 사용할 수 없는 문제로 보여집니다.

아래 샘플 코드 참고 부탁 드립니다.
https://github.com/naverz/zepeto-world-sample/blob/main/Assets/Chapter1/Scripts/Teleport.ts

**Reply by choramha** [reply]
답변 감사합니다. 샘플 코드 참고해서 수정한 결과, OnTriggerEnter 에 텔레포트 함수를 넣었을땐 정상적으로 작동하지만, onClick에 텔레포트 함수를 넣었을땐 작동하지가 않네요... 왜그럴까요 ㅜㅜ

**Reply by YSHFM** [reply]
AddListner 콜백 함수 호출 되는 것을 확인도 하시고, 마찬가지로 zepetoCharacter가 null이시라면, 
UI내에 onEnable , onDisable함수등을 이용해서 해보는 것도 방법이실 것 같습니다. 콜백함수로 enable disable컨트롤을 해주고 Enable될 때에 zepetoCharacter을 getPlayer등을 해서 텔레포트 하셔도 될 것 같네요.

**Reply by choramha** [reply]
감사합니다. 해결했습니다!
